pennylane_lightning_gpu/src/algorithms/ObservablesGPUMPI.hpp

```diff
@@ -404,4 +406,294 @@ class HamiltonianGPUMPI final : public ObservableGPUMPI<T> {
     }
 };
 
+/**
+ * @brief Sparse representation of HamiltonianGPUMPI<T>
+ *
+ * @tparam T Floating-point precision.
+ */
+template <typename T>
+class SparseHamiltonianGPUMPI final : public ObservableGPUMPI<T> {
+  public:
+    using PrecisionT = T;
```

Comment:
Could we template directly on `PrecisionT`?

Reply:
yea, current implementation is aligned with Lightning, non-MPI backend of Lightning-GPU. I can update all when we integrate them into a same repo.